src/compiler/moduleSpecifiers.ts

```diff
@@ -453,6 +457,23 @@ namespace ts.moduleSpecifiers {
                 else if (pattern === relativeToBaseUrl || pattern === relativeToBaseUrlWithIndex) {
                     return key;
                 }
+                else if (resolvedPaths) {
+                    const resolvedCandidate = resolvedPaths?.[key]?.[i];

```

Comment:
You shouldn't need the `?.` after `resolvedPaths` given that you checked it above, right?